File: CinemaReservations/CSharp/CinemaAllocations.Domain/SeatAllocation.cs
using System.Collections.Generic;

namespace CinemaAllocations.Domain
{
    public class SeatAllocation
    {
        public int PartyRequested { get; }
      
        public List<Seat> AllocatedSeats { get; } = new List<Seat>();
     
        public bool IsFulfilled => AllocatedSeats.Count == PartyRequested;

        public SeatAllocation(int partyRequested)
        {
            PartyRequested = partyRequested;
        }

        public void AddSeat(Seat seat)
        {
            AllocatedSeats.Add(seat);
        }
    }
}

File: CinemaReservations/Java/CinemaAllocationsDomain/src/main/java/com/sdd/cinemaallocations/SeatAllocation.java
package com.sdd.cinemaallocations;

import lombok.Value;

import java.util.ArrayList;
import java.util.List;

@Value
public class SeatAllocation {

    public int partyRequested;
    public List<Seat> allocatedSeats = new ArrayList<>();

    public boolean isFulfilled() {
        return allocatedSeats.size() == partyRequested;
    }

    public SeatAllocation(int partyRequested)
    {
        this.partyRequested = partyRequested;
    }
    public void addSeat(Seat seat)
    {
        allocatedSeats.add(seat);
    }
}